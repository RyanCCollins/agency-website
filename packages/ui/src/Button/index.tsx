import * as React from 'react';
import ButtonComponent from './styles';
import { Size } from './types';

// This is so that the onClick handler is accepted without type interferance
export interface Props extends React.HTMLProps<HTMLButtonElement & Button> {
  color?: string;
  backgroundColor?: string;
  fontSize?: Size;
  borderColor?: string;
  label?: string;
}

class Button extends React.Component<Props, undefined> {
  public static defaultProps: Props = {
    color: '#fefefe',
    backgroundColor: '#c05b4d',
    borderColor: '#732419',
    fontSize: 'medium',
  };
  public render() {
    const { children, label, ...props } = this.props;
    return (
      <ButtonComponent
        {...props}
      >
        {children || label || ''}
      </ButtonComponent>
    );
  }
}

export default Button;
