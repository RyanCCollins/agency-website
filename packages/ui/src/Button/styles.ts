import styled from 'styled-components';
import remStringFromPX from '../utils';
import { sizeMap } from './maps';
import { Props } from './';

export default styled.button`
  text-align: center;
  background-color: ${(props: Props) => props.backgroundColor};
  color: ${(props: Props) => props.color};
  font-size: ${(props: Props) => remStringFromPX(sizeMap[props.fontSize])};
  border: ${(props: Props) => props.borderColor ? `3px solid ${props.borderColor}` : 'none'};
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  cursor: pointer;
`;
