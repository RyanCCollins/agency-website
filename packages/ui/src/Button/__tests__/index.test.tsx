import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Button from '../';
import { buttonProps } from './__mocks__/buttonMocks.mock';

describe('<Button /> default', () => {
  it('should render a plain button with default props', () => {
    const wrapper = shallow(
      <Button>
        Button Text
      </Button>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render a plain button with other props', () => {
    const wrapper = shallow(
      <Button {...buttonProps}>
        Button Text
      </Button>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('onClick event for plain button', () => {
  let mockFunction;
  let button;

  beforeEach(() => {
    mockFunction = jest.fn();
    button = mount(
      <Button onClick={mockFunction}>
        Button Text
      </Button>,
    );
  });

  it('Button requires onClick prop', () => {
    expect(button.props().onClick).toBeDefined();
  });

  it('Button renders button', () => {
    const renderedButton = button.find('button').first();
    expect(renderedButton).toBeDefined();
  });

  it('Button click calls mockFunction', () => {
    const renderedButton = button.find('button').first();
    renderedButton.simulate('click');
    expect(mockFunction).toBeCalled();
  });
});
