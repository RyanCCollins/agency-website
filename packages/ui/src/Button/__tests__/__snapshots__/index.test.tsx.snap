// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> default should render a plain button with default props 1`] = `
<styled.button
  backgroundColor="#c05b4d"
  borderColor="#732419"
  color="#fefefe"
  fontSize="medium"
>
  Button Text
</styled.button>
`;

exports[`<Button /> default should render a plain button with other props 1`] = `
<styled.button
  backgroundColor="#ffffff"
  borderColor="#000000"
  color="blue"
  fontSize="medium"
  size={300}
>
  Button Text
</styled.button>
`;
